src/visitor.rs

```diff
@@ -524,6 +524,11 @@ pub trait Visitor<'a> {
                     )
                 }
             }
+            FunctionType::Cast(cast) => format!(
+                "CAST({expr} AS {tpe})",
+                expr = self.visit_database_value(*cast.expr),
+                tpe = cast.tpe
```

Comment:
Not sure we manipulate the type as a raw string. I haven't looked extremely in depth but I haven't found a wrapper for types in the AST.

Reply:
Should you parameterize the `tpe` and did you try it out in a database query? Maybe using an enum for all possible cast types...